project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,386 @@
+            type: "ir.actions.act_window",
+            domain: [
+                ["account_id.project_ids", "=", this.projectId],
+                ["product_id.type", "in", ["consu", "product"]],
+                ["product_id.categ_id", "=", categoryId],
+            ],
+        });
+    },
+    /**
+     * Handle a click on the label of a time category.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    timeCategoryClicked(event){
+        event.preventDefault();
+        var categoryId = getRecordIdFromEvent(event);
+        this.categoryClicked(categoryId, 'time');
+    },
+    /**
+     * Handle a click on the total of a time category.
+     *
+     * The list of analytic lines contained in the category is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    hourTotalClicked(event){
+        event.preventDefault();
+        var taskTypeId = getRecordIdFromEvent(event);
+        var taskTypeName = getRecordNameFromEvent(event);
+        var actionName = _t("Analytic Lines ({category})").replace("{category}", taskTypeName);
+        this.do_action({
+            name: actionName,
+            res_model: "account.analytic.line",
+            views: [[false, "list"], [false, "form"]],
+            type: "ir.actions.act_window",
+            domain: [
+                ["account_id.project_ids", "=", this.projectId],
+                ["task_id.task_type_id", "=", taskTypeId],
+                ["task_id", "!=", false],
+            ],
+        });
+    },
+    /**
+     * Handle a click on the label of an uotsourcing category.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    outsourcingCategoryClicked(event){
+        event.preventDefault();
+        var categoryId = getRecordIdFromEvent(event);
+        this.categoryClicked(categoryId, 'outsourcing');
+    },
+    /**
+     * Handle a click on the total of an outsourcing category.
+     *
+     * The list of analytic lines contained in the category is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    outsourcingTotalClicked(event){
+        event.preventDefault();
+        var categoryId = getRecordIdFromEvent(event);
+        var categoryName = getRecordNameFromEvent(event);
+        var actionName = _t("Analytic Lines ({category})").replace("{category}", categoryName);
+        this.do_action({
+            name: actionName,
+            res_model: "account.analytic.line",
+            views: [[false, "list"], [false, "form"]],
+            type: "ir.actions.act_window",
+            domain: [
+                ["account_id.project_ids", "=", this.projectId],
+                ["task_id", "=", false],
+                ["product_id.type", "=", "service"],
+            ],
+        });
+    },
+    /**
+     * Show the widget.
+     *
+     * This is a standard method of web.Widget.
+     * It is called when the report is reopened after a click on the breadcrumb.
+     */
+    do_show(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'do_show' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)